File: 199. Binary Tree Right Side View/Solution.java
/**
 * https://leetcode.com/problems/binary-tree-right-side-view/?envType=study-plan-v2&envId=leetcode-75
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        var li = new ArrayList<Integer>();

        if(root  == null)
            return li;
        
        Queue <TreeNode> q = new LinkedList<>();
        q.add(root);

        while(!q.isEmpty()){
            int size = q.size();
            int val = 0;
            for(int i = 0; i < size; i++){
                var node = q.poll();
                val = node.val;

                if(node.left  != null){
                    q.add(node.left);
                }
                if(node.right != null){
                    q.add(node.right);
                }
            }
            li.add(val);
        }
        return li;
    }
}

File: 199. Binary Tree Right Side View/Solution.py
##https://leetcode.com/problems/binary-tree-right-side-view/?envType=study-plan-v2&envId=leetcode-75

class TreeNode(object):
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution(object):
    def rightSideView(self, root):
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        lis = []
        level  = 0
        self.dfs(root,lis,level)
        return lis
        
    def dfs(self, node, li , level):
        if not node:
            return None
        
        if len(li) == level:
            li.append(node.val)

        self.dfs(node.right, li, level + 1)
        self.dfs(node.left, li, level + 1)
        
        